DiscordBot/Commands/EventCommands.cs

```diff
@@ -90,6 +99,117 @@ public async Task CreateEvent(CommandContext context, InteractivityModule intera
             await context.RespondAsync(embed: embed);
         }
 
+        public async Task EditEvent(CommandContext context, InteractivityModule interactivity)
+        {
+            await context.RespondAsync($"{context.Member.Mention} - what is the event key?");
+            var eventKey = await GetUserIntResponse(context, interactivity);
+            if (eventKey == null)
+            {
+                return;
+            }
+
+            var embed = await GetEventEmbed(context, eventKey.Value);
+            if (embed == null)
+            {
+                return;
+            }
+
+            await EditEventFields(context, interactivity, eventKey.Value, embed);
+        }
+
+        private static async Task EditEventFields(
+            CommandContext context,
+            InteractivityModule interactivity,
+            int eventKey,
+            DiscordEmbedBuilder embed)
+        {
+            string? newName = null;
+            string? newDescription = null;
+            DateTimeOffset? newTime = null;
+
+            while (true)
```

Comment:
I did want to wrap this loop around the call to this function from EditEvent, but would then need to add some kind of return value in order to terminate the loop from outside of this method.

Would that be preferable?

Reply:
I'm not sure a loop is needed. What about instead asking the user:
"Enter new name for event (leave blank if you want to keep current name):"
"Enter new time for event (leave blank ...)"
and so on until you have all the fields you need. What do you think? 🙂

Reply:
Discord doesn't support blank responses so that approach would require some keyword to skip each field, which I would prefer not to do.

I don't suppose recursion is an option?

Reply:
I don't think recursion would make the code any clearer. I'm just not sure an infinite loop is the right solution, both from a code perspective and from a user experience perspective. Because that would let them change e.g. the name again and again. If blank responses aren't supported then what about:
"Do you want to change the name, enter y/n"
if y:
"What's the new name?"
"Do you want to change the time?"
...

What do you think? 🙂

Reply:
> Because that would let them change e.g. the name again and again.
I don't think this is necessarily a bad thing, since it allows the user to change their mind.
I could use while (!isFinished) then assign that value inside the loop? Would that be an improvement?

Reply:
They can always change their mind by just editing the event again. No, having and `isFinished` variable wouldn't really make any difference as the issue is with nesting everything thing in a loop. I think we're making the code more complicated for behaviour which isn't really necessary.

But it's your bot so you can have that behaviour if you want. In that case you need to split up this function as it's very long with several levels of nesting which makes it hard to read.